cache/in-memory/src/permission.rs

```diff
@@ -0,0 +1,748 @@
+
+        let overwrites = match &channel.data {
+            GuildChannel::Category(c) => &c.permission_overwrites,
+            GuildChannel::Stage(c) => &c.permission_overwrites,
+            GuildChannel::Text(c) => &c.permission_overwrites,
+            GuildChannel::Voice(c) => &c.permission_overwrites,
+        };
+
+        let calculator =
+            PermissionCalculator::new(guild_id, user_id, everyone, assigned.as_slice());
+
+        Ok(calculator.in_channel(channel.data.kind(), overwrites))
+    }
+
+    /// Calculate the guild-level permissions of a member.
+    ///
+    /// Returns [`Permissions::all`] if the user is the owner of the guild.
+    ///
+    /// The following [`ResourceType`]s must be enabled:
+    ///
+    /// - [`ResourceType::MEMBER`]
+    /// - [`ResourceType::ROLE`]
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// use twilight_cache_inmemory::InMemoryCache;
+    /// use twilight_model::id::{GuildId, UserId};
+    ///
+    /// let cache = InMemoryCache::new();
+    ///
+    /// // later on...
+    ///
+    /// let guild_id = GuildId(4);
+    /// let user_id = UserId(5);
+    ///
+    /// let permissions = cache.permissions().root(user_id, guild_id)?;
+    /// println!(
+    ///     "User {} in guild {} has permissions {:?}",
+    ///     user_id,
+    ///     guild_id,
+    ///     permissions,
+    /// );
+    /// # Ok(()) }
+    /// ```
+    ///
+    /// # Errors
+    ///
+    /// Returns a [`RootErrorType::MemberUnavailable`] error type if the
+    /// member for the user in the guild is not present.
+    ///
+    /// Returns a [`RootErrorType::RoleUnavailable`] error type if one of the
+    /// member's roles is not in the cache.
+    ///
+    /// [`Permissions::all`]: twilight_model::guild::Permissions::all
+    /// [`ResourceType::MEMBER`]: crate::ResourceType::MEMBER
+    /// [`ResourceType::ROLE`]: crate::ResourceType::ROLE
+    /// [`ResourceType`]: crate::ResourceType
+    pub fn root(&self, user_id: UserId, guild_id: GuildId) -> Result<Permissions, RootError> {
+        if self.is_owner(user_id, guild_id) {
+            return Ok(Permissions::all());
+        }
+
+        let MemberRoles { assigned, everyone } = self
+            .member_roles(user_id, guild_id)
+            .map_err(RootError::from_member_roles)?;
+        let calculator =
+            PermissionCalculator::new(guild_id, user_id, everyone, assigned.as_slice());
+
+        Ok(calculator.root())
+    }
+
+    /// Determine whether a given user is the owner of a guild.
+    ///
+    /// Returns true if the user is or false if the user is definitively not the
+    /// owner of the guild or the guild is not in the cache.
+    fn is_owner(&self, user_id: UserId, guild_id: GuildId) -> bool {
+        (self.0)
+            .0
+            .guilds
+            .get(&guild_id)
+            .map(|r| r.owner_id == user_id)
+            .unwrap_or_default()
```

Comment:
`unwrap_or(false)` is clearer

Reply:
I don't agree. The default value of a boolean is clearly not `true` and Clippy promotes the use of `unwrap_or_default`. We also use it elsewhere in the codebase.